### PR TITLE
More windows build work (WIP)

### PR DESCRIPTION
**Pull request by tpetr**
- better utilize makefile for windows build
- make Common2.glob more portable
- remove hardcoded unix-style paths

Fixed a bunch of these with this Semgrep rule:
```yaml
rules:
  - id: fix-glob
    patterns:
    - pattern-either:
      - pattern: |
          spf $FORMAT $X
      - pattern: |
          Common.spf $FORMAT $X
    - metavariable-regex:
        metavariable: $FORMAT
        regex: '"%s/(?P<GLOB>.*)"'
    - pattern-inside: |
        Common2.glob ...
    message: Use filename.concat instead
    fix: |
      (Filename.concat $X "$GLOB")
    languages: [ocaml]
    severity: WARNING
```

### CODE REVIEW

**Comment by tpetr** [comment]
[Progress](https://github.com/semgrep/semgrep/actions/runs/7864666832/job/21456700062)... I'm sure at least a handful of these are brittle path comparisons
```
1569/1569 selected tests:
  1535 successful (1533 pass, 2 xfail)
  34 unsuccessful (34 fail, 0 xpass)
overall status: failure
```

**Comment by aryx** [comment]
cc @mjambon who made some transition to use Fpath.t who has a bit of experience of Windows vs Unix path.

File: .github/workflows/libs/gha.libsonnet [hunk 1]
```diff
@@ -55,6 +55,12 @@
     run: 'git config --system core.longpaths true'
   },
 
+  git_set_core_autocrlf(mode): {
```
Comment (aryx) [inline .github/workflows/libs/gha.libsonnet:58]: comment please, why we need that (can say it's for windows, with a vague explanation
of why we need it).

File: .github/workflows/libs/gha.libsonnet [hunk 2]
```diff
@@ -55,6 +55,12 @@
...
+      run: |||
```
Comment (aryx) [inline .github/workflows/libs/gha.libsonnet:59]: you can also use it on one line, run: "foobar %s " % mode.

**Review by aryx — APPROVED** [review]
Very nice!
Feel free to merge once the CI tests are all passing.
Also you can extend semgrep.jsonnet with the rule you wrote to help assist you,
so we prevent new use if spf "%s/..." to get in.

**Pull request closed without merging**